### ISSUE TITLE
Navtree - report function properties so that they can be identified as functions

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#25062]
https://github.com/Microsoft/vscode/issues/51988

**TypeScript Version:**  3.0.0-dev.20180616

**Search Terms:** 
- navtree
- method, function, property

**Code**
For the code:
```ts
class K {
    p = 1
    f = () => {}
    m() {}
}
```

Run a `navtree` request

**Expected behavior:**
Some way to identify that `f` is a function

**Actual behavior:**
Both `p` and `f` are reported the same way, as properties.

```
Result: {
    "text": "<global>",
    "kind": "script",
    "kindModifiers": "",
    "spans": [
        {
            "start": {
                "line": 1,
                "offset": 1
            },
            "end": {
                "line": 5,
                "offset": 3
            }
        }
    ],
    "childItems": [
        {
            "text": "K",
            "kind": "class",
            "kindModifiers": "",
            "spans": [
                {
                    "start": {
                        "line": 1,
                        "offset": 1
                    },
                    "end": {
                        "line": 5,
                        "offset": 2
                    }
                }
            ],
            "nameSpan": {
                "start": {
                    "line": 1,
                    "offset": 7
                },
                "end": {
                    "line": 1,
                    "offset": 8
                }
            },
            "childItems": [
                {
                    "text": "f",
                    "kind": "property",
                    "kindModifiers": "",
                    "spans": [
                        {
                            "start": {
                                "line": 3,
                                "offset": 5
                            },
                            "end": {
                                "line": 3,
                                "offset": 17
                            }
                        }
                    ],
                    "nameSpan": {
                        "start": {
                            "line": 3,
                            "offset": 5
                        },
                        "end": {
                            "line": 3,
                            "offset": 6
                        }
                    }
                },
                {
                    "text": "m",
                    "kind": "method",
                    "kindModifiers": "",
                    "spans": [
                        {
                            "start": {
                                "line": 4,
                                "offset": 5
                            },
                            "end": {
                                "line": 4,
                                "offset": 11
                            }
                        }
                    ],
                    "nameSpan": {
                        "start": {
                            "line": 4,
                            "offset": 5
                        },
                        "end": {
                            "line": 4,
                            "offset": 6
                        }
                    }
                },
                {
                    "text": "p",
                    "kind": "property",
                    "kindModifiers": "",
                    "spans": [
                        {
                            "start": {
                                "line": 2,
                                "offset": 5
                            },
                            "end": {
                                "line": 2,
                                "offset": 10
                            }
                        }
                    ],
                    "nameSpan": {
                        "start": {
                            "line": 2,
                            "offset": 5
                        },
                        "end": {
                            "line": 2,
                            "offset": 6
                        }
                    }
                }
            ]
        }
    ]
}

```

**Related Issues:** 
- #14830

**Comment by mhegazy** [comment]
@andy-ms we did try this in the past. i can not remember what were the issues we ran into.. do you recall?

**Comment by andy-ms** [comment]
I don't remember trying this before. One caveat `navTree` doesn't use the type checker so we would have to specifically be looking for an arrow function node, rather than supporting any callable type. But if you prefer it, we could report `f` as a method instead of a property, or add a new `ScriptElementKindModifier.callable`.

**Comment by mhegazy** [comment]
i think that was the issue... sometimes we would report it as a functions and sometimes we would not..

**Comment by mhegazy** [comment]
I think we are better off keeping it consistent.

**Comment by mjbvz** [comment]
Sounds good. We can always revisit this if we start hearing about this more often
